### ISSUE TITLE
Support File as Request Body

### ISSUE THREAD

**Issue by animator** [foss42/apidash#352]
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
Currently API Dash supports `text`, `json`, `multipart formdata` as request body.
Support should also be provided to attach just a single file.
The contents of the file is sent as bytes and the content-type is `application/octet-stream`

**Comment by AcousticDeveloper** [comment]
Shall we add one more tab like section named `File Upload` beside the existing three and it can have a drag and drop interface? Or for initial setup, the existing file upload option's `select file` is preferred? 

![image](https://github.com/foss42/apidash/assets/161834431/3fa1d394-25f7-49f8-8052-29ed1c6ed424)

**Comment by Tanish2002** [comment]
@animator would this include rewriting all codegens and their corresponding tests as well?

Also, I believe I would have to add a new state variable to `class RequestModel` to store the file name, I can't simply use `requestBody` since it's shared by json and text type requests and when switching between them, it would conflict.

**Comment by alainjr10** [comment]
Has anyone noticed file picker not working on linux? I don't know if i should raise a new issue or maybe my environment isn't properly configured.
The pick file method returns null

**Comment by Tanish2002** [comment]
> Has anyone noticed file picker not working on linux? I don't know if i should raise a new issue or maybe my environment isn't properly configured. The pick file method returns null

@alainjr10 Yeah this isn't documented but on linux you need `zenity` installed to make it work correctly. 


@animator I believe the mpv and zenity dependency should be documented for linux developers

**Comment by alainjr10** [comment]
> > Has anyone noticed file picker not working on linux? I don't know if i should raise a new issue or maybe my environment isn't properly configured. The pick file method returns null
> 
> @alainjr10 Yeah this isn't documented but on linux you need `zenity` installed to make it work correctly. 
> 
> 
> @animator I believe the mpv and zenity dependency should be documented for linux developers
> 
> Edit: Link to the package on repology: https://repology.org/project/zenity/versions
> 
> Edit 2: it needs one of the following `qarma`, `kdialog`, `zenity`:
> source: https://github.com/miguelpruivo/flutter_file_picker/blob/04e32daa8180e4d93a96c1d0ffba9d61feb25b10/lib/src/linux/file_picker_linux.dart#L108


Thanks a lot @Tanish2002 .
I been stuck on this issue since yesterday. 
I'll try it and give you feedback.

**Comment by Tanish2002** [comment]
FWIW, I've opened a PR to potentially fix this issue #357
